### PR TITLE
add missing @return tags for Array.prototype.sort() and Array.prototype.reverse()

### PR DESCRIPTION
**Pull request by adufilie**

### CODE REVIEW

**Comment by adufilie** [comment]
I signed it.

File: externs/es3.js [hunk 1]
```diff
@@ -513,6 +513,7 @@ Array.prototype.push = function(var_args) {};
  * Transposes the elements of an array in place: the first array element becomes the
  * last and the last becomes the first.
  *
+ * @return {!Array<T>}
```
Comment (Dominator008) [inline externs/es3.js]: This will need a "@template T"

File: externs/es3.js [hunk 2]
```diff
@@ -513,6 +513,7 @@ Array.prototype.push = function(var_args) {};
...
  * @this {{length: number}}
  * @modifies {this}
  * @see http://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reverse
```
Comment (Dominator008) [inline externs/es3.js]: While we are here, can you fix this URL to be http://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice ?

Comment (Dominator008) [inline externs/es3.js]: While we are here, can you fix this URL to be http://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice ?

**Comment by adufilie** [comment]
I've added the template T, but regarding the URLs they seem to be correct.

**Comment by Dominator008** [comment]
@adufilie I misread the URL on my phone :(

**Comment by dimvar** [comment]
FYI, these changes will probably require cleaning up code internally, and if the cleanup required ends up being big, the PR may get reverted.

**Comment by Dominator008** [comment]
@dimvar Acknowledged.

**Comment by blickly** [comment]
In order to get back to a passing state with both repositories in sync so we can rerelease, I'm reverting this change for now.

**Comment by Dominator008** [comment]
@blickly Thanks!

**Comment by blickly** [comment]
It looks like a good type-checking improvement, but unfortunately that makes it more difficult to land today.

**Pull request merged**